### PR TITLE
docs(holepunching-tutorial): update to new example names

### PR DESCRIPTION
**Pull request by dvorakchen**
## Description

punching tutorial code cannot work 
Related https://github.com/libp2p/rust-libp2p/issues/3911

in [hole punching tutorial](https://docs.rs/libp2p/latest/libp2p/tutorials/hole_punching/index.html) example code not work

example code in examples directory in root directory, cannot execute by following:
`cargo build --example relay_v2 -p libp2p-relay`

That's confused, so I created `examples` directory in examples directory of root, let it can execute by
`cargo build --example relay_v2`

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: libp2p/src/tutorials/hole_punching.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@
 //!
 //! ``` bash
 //! ## Inside the rust-libp2p repository.
-//! cargo build --example relay_v2 -p libp2p-relay
+//! cargo build --example relay_v2
```
Comment (mxinden) [inline libp2p/src/tutorials/hole_punching.rs:57]: I think this has simply been renamed to `relay-server-example`. Does that work for you?

File: libp2p/src/tutorials/hole_punching.rs [hunk 2]
```diff
@@ -122,7 +122,7 @@
 //!
 //! ``` bash
 //! ## Inside the rust-libp2p repository.
-//! cargo build --example client -p libp2p-dcutr
+//! cargo build --example client
```
Comment (mxinden) [inline libp2p/src/tutorials/hole_punching.rs:125]: Renamed to `dcutr` as far as I can tell.

File: libp2p/src/tutorials/hole_punching.rs (hunk 1 above)
Comment (dvorakchen) [reply libp2p/src/tutorials/hole_punching.rs:57]: `cargo build --example relay_v2 -p libp2p-relay` did not work,
I got this tip: `error: no example target named 'relay_v2'.`

remove ` -p libp2p-relay` would be work

File: libp2p/src/tutorials/hole_punching.rs (hunk 2 above)
Comment (dvorakchen) [reply libp2p/src/tutorials/hole_punching.rs:125]: hole punching tutorial in website: [libp2p::tutorials::hole_punching - Rust](https://docs.rs/libp2p/latest/libp2p/tutorials/hole_punching/index.html#setting-up-the-listening-client)

example code still be 
`## Inside the rust-libp2p repository.
cargo build --example client -p libp2p-dcutr`

so... were the docs not updated?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Let's not introduce duplicate code here and instead update the tutorial to use the correct example names :)

Thank you!

File: libp2p/src/tutorials/hole_punching.rs (hunk 2 above)
Comment (mxinden) [reply libp2p/src/tutorials/hole_punching.rs:125]: > so... were the docs not updated?

Very likely, yes.

**Pull request closed without merging**